src/main/java/qupath/ext/imglib2/imageserver/ImgLib2ImageServer.java

```diff
@@ -0,0 +1,418 @@
+                        int c = position[ImgCreator.AXIS_CHANNEL] - minTile[ImgCreator.AXIS_CHANNEL];
+                        int xy = position[ImgCreator.AXIS_X] - minTile[ImgCreator.AXIS_X] +
+                                (position[ImgCreator.AXIS_Y] - minTile[ImgCreator.AXIS_Y]) * tileRequest.getTileWidth();
+
+                        pixels[c][xy] = value.get();
+                    }
+
+                    yield new DataBufferFloat(pixels, xyPlaneSize);
+                }
+                case FLOAT64 -> {
+                    double[][] pixels = new double[numberOfChannelsInAccessibles][xyPlaneSize];
+
+                    while (cursor.hasNext()) {
+                        DoubleType value = (DoubleType) cursor.next();
+
+                        cursor.localize(position);
+                        int c = position[ImgCreator.AXIS_CHANNEL] - minTile[ImgCreator.AXIS_CHANNEL];
+                        int xy = position[ImgCreator.AXIS_X] - minTile[ImgCreator.AXIS_X] +
+                                (position[ImgCreator.AXIS_Y] - minTile[ImgCreator.AXIS_Y]) * tileRequest.getTileWidth();
+
+                        pixels[c][xy] = value.get();
+                    }
+
+                    yield new DataBufferDouble(pixels, xyPlaneSize);
+                }
+            };
+
+            return new BufferedImage(
+                    ColorModelFactory.createColorModel(metadata.getPixelType(), metadata.getChannels()),
+                    WritableRaster.createWritableRaster(
+                            new BandedSampleModel(
+                                    dataBuffer.getDataType(),
+                                    tileRequest.getTileWidth(),
+                                    tileRequest.getTileHeight(),
+                                    numberOfChannelsInAccessibles
+                            ),
+                            dataBuffer,
+                            null
+                    ),
+                    false,
+                    null
+            );
+        }
+    }
+
+    @Override
+    protected ImageServerBuilder.ServerBuilder<BufferedImage> createServerBuilder() {
+        return new ImgLib2ServerBuilder<>(accessibles, getMetadata());
+    }
+
+    @Override
+    protected String createID() {
+        return id;
+    }
+
+    @Override
+    public Collection<URI> getURIs() {
+        return List.of();
+    }
+
+    @Override
+    public String getServerType() {
+        return "ImgLib2";
+    }
+
+    @Override
+    public ImageServerMetadata getOriginalMetadata() {
+        return metadata;
+    }
+
+    @Override
+    protected BufferedImage createDefaultRGBImage(int width, int height) {
+        return new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
+    }
+
+    private static <T extends NativeType<T> & NumericType<T>> void checkInputs(List<RandomAccessibleInterval<T>> accessibles, ImageServerMetadata metadata) {
+        for (RandomAccessibleInterval<T> accessible: accessibles) {
```

Comment:
Check here for null or empty list, and throw a more informative exception if found?